### DISCUSSION TITLE
Node-RED OpenAI API Version 1.5.0 Released!

### DISCUSSION THREAD

**Discussion by allanbunch** [allanbunch/node-red-openai-api#26] [Announcements]
### Small change; big deal!

With the release of the [OpenAI Realtime API](https://openai.com/index/introducing-the-realtime-api/), exciting new possibilities are now available to the Node-RED community. Whether you’ve been itching to explore advanced AI voice-assisted ideas or something entirely new, the OpenAI team has made it possible.

It’s been a couple of weeks since the Realtime API’s initial beta release, and now OpenAI is rolling out some `gpt-4o-audio-preview` model features into their official SDKs. This is great news because one of the core design principles of the Node-RED OpenAI API node is seamless parity with the underlying [OpenAI Node API Library](https://www.npmjs.com/package/openai). Put simply—it just works.

With version 1.5.0 of the Node-RED OpenAI API node, you now have access to the `gpt-4o-audio-preview` [audio generation](https://platform.openai.com/docs/guides/audio) features. Usage follows the same pattern as a standard chat completion request. Here’s how you can get started right away:

```javascript
// Node-RED function node code example
msg.payload = {
  "model": "gpt-4o-audio-preview",
  "modalities": ["text", "audio"],
  "audio": { voice: "alloy", format: "wav" },
  "messages": [
    {
      "role": "user",
      "content": "Is a golden retriever a good family dog?"
    }
  ]
}
return msg;
```

And for streaming responses:

```javascript
// Node-RED function node code for streamed response
msg.payload = {
  "model": "gpt-4o-audio-preview",
  "stream": true,
  "modalities": ["text", "audio"],
  "audio": { voice: "alloy", format: "pcm16" },
  "messages": [
    {
      "role": "user",
      "content": "Is a golden retriever a good family dog?"
    }
  ]
}
return msg;
```

Stay tuned for more updates! I’ll continue incorporating new features into the node as they roll out in the official OpenAI Node API Library.

**Comment by riekusr** [comment]
Can't wait to try it out!
